client/app/api/course/MaterialFolders.js

```diff
@@ -0,0 +1,26 @@
+import BaseCourseAPI from './Base';
+
+export default class MaterialFoldersAPI extends BaseCourseAPI {
+  /**
+  * Upload files to the specified folder.
+  *
+  * @param {number} folderId
+  * @param {array} files - A list of files from file input.
+  * @return {Promise}
+  * success response: { materials: Array.<{id:number, name:string, url:string, updated_at:string}> }
```

Comment:
@kxmbrian I changed to use the JSDoc way to document the return object, this was we should also be able to document complex object. http://usejsdoc.org/tags-type.html